cmd/juju/application/deployer/bundle.go

```diff
@@ -177,6 +179,25 @@ Please repeat the deploy command with the --trust argument if you consent to tru
 	return nil
 }
 
+func (d *deployBundle) printDryRunUnmarshalErrors(ctx *cmd.Context, unmarshalErrors []error) {
+	if !d.dryRun {
+		return
+	}
+	// During a dry run, print any unmarshalling errors from the
+	// bundles and overlays
+	var msg string
+	for _, err := range unmarshalErrors {
+		if err == nil {
+			continue
+		}
+		msg = fmt.Sprintf("%s\n %s\n", msg, err)
+	}
+	if msg == "" {
+		return
+	}
+	ctx.Warningf("Fields failing unmarshal will be ignored at deployment.%s\n", msg)
```

Comment:
I don't know that users will understand "unmarshal". Maybe just something like "These fields [...] will be ignored during deployment"